src/ol/source/Image.js

```diff
@@ -127,10 +127,9 @@ class ImageSource extends Source {
 
     /**
      * @private
-     * @type {Array<number>}
+     * @type {Array<number>|undefined}
```

Comment:
Why `undefined` instead of `null` here?

Reply:
Because then it was an easy match with the `getResolutions` defined in the parent class https://github.com/openlayers/openlayers/blob/main/src/ol/source/Source.js#L155

Reply:
If you instead changed the type of the return on the parent to `Array<string>|null`, then you wouldn't need to change the code below.

In addition, `Array<string>|null` would be consistent with the other type annotation changes you are making here.

I think it would be preferable if this pull request were restricted to type annotation changes (comments) instead of code changes.  If there are bugs revealed in changing the type annotations, then that would justify code changes.  Otherwise, including unnecessary code changes is more likely to introduce new bugs.

Reply:
I agree null would fit closer to the code style here.
Should we leave https://github.com/openlayers/openlayers/blob/main/src/ol/View.js#L449 like it is, or also switch undefined for null, but then requiring a code change again for the option parsing (which is internal so I think it's not a problem)

Reply:
I'd be tempted to see how far you can get with comment-only changes toward supporting `strictNullChecks: true`.  So not changing any of the behavior, just the type checking.

Later we could work on making things consistently `Something|null` or `Something|undefined`.   I guess my preference would be to choose whatever minimizes unnecessary guards, ternary expressions, and extra code.  It feels like `Something|undefined` would be simpler.  But I think this would probably be a breaking change.

Anybody else have any opinions?

Reply:
Totally agree. It means reviewing the whole code base to see where we actually allow `null` and where we don't. Regarding `null` vs `undefined`, I think we have several places in the code where `null` has a different effect than `undefined`. So there will definitely be breaking changes, but I agree that the old Closure Library advice to use `null` for `Object`s and `undefined` for simple types when they are not set may not hold true any more.